clusterloader2/framework/util/util.go

```diff
@@ -0,0 +1,201 @@
+func DeleteNamespace(c clientset.Interface, namespace string) error {
+	deleteFunc := func() (bool, error) {
+		err := c.CoreV1().Namespaces().Delete(namespace, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete namespace with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+// CreateObject creates object based on given object description.
+func CreateObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind, obj *unstructured.Unstructured) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	obj.SetName(name)
+	createFunc := func() (bool, error) {
+		_, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(obj)
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to create object with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(createFunc)
+}
+
+// DeleteObject deletes object with given name, group, version and kind.
+func DeleteObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	deleteFunc := func() (bool, error) {
+		err := dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete object with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+// GetObject retrieves object with given name, group, version and kind.
+func GetObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) (*unstructured.Unstructured, error) {
+	var obj *unstructured.Unstructured
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	getFunc := func() (bool, error) {
+		var err error
+		obj, err = dynamicClient.Resource(gvr).Namespace(namespace).Get(name, metav1.GetOptions{
+			IncludeUninitialized: true,
+		})
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to retrieve object with non-retryable error: %v", err)
+	}
+	if err := RetryWithExponentialBackOff(getFunc); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+// UpdateObject updates with given name, group, version and kind based on given object description.
+func UpdateObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind, obj *unstructured.Unstructured) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	obj.SetName(name)
+	currentObj, err := GetObject(dynamicClient, namespace, name, gvk)
+	if err != nil {
+		return err
+	}
+	current, err := currentObj.MarshalJSON()
```

Comment:
I don't get this - we don't want to create patch and send patch requests. Patch may be a separate command - we want to use Update.

DynamicClient exposes Update method - there are places where this is used, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/operations.go#L76

This method should look exactly the same way as create and delete.

Reply:
update reimplemented.